youtube_dl/extractor/infomaniak.py

```diff
@@ -0,0 +1,88 @@
+# coding: utf-8
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+from ..utils import (
+    ExtractorError,
+    parse_duration,
+)
+
+
+class InfomaniakVod2IE(InfoExtractor):
+    IE_NAME = 'infomaniak:vod2'
+    _VALID_URL = r'https?://player\.vod2\.infomaniak\.com/embed/(?P<id>[0-9a-z]+)'
+    _TEST = {
+        'url': 'https://player.vod2.infomaniak.com/embed/1jhvl2uqg6ywp',
+        'info_dict': {
+            'id': '1jhvl2uqg6xis',
+            'display_id': '1jhvl2uqg6ywp',
+            'ext': 'mp4',
+            'title': 'Conférence à Dyo, octobre 2022',
+            'thumbnail': r're:^https?://.*\.(?:jpe?g|png)$',
+        },
+        'params': {
+            'skip_download': True,
+        },
+    }
+
+    def _real_extract(self, url):
+        share_id = self._match_id(url)
+        share = self._download_json(
+            'https://api.infomaniak.com/2/vod/res/shares/%s.json' % share_id,
+            share_id)
+        data = (share or {}).get('data') or {}
+        media = data.get('media') or []
+
+        entries = []
+        for m in media:
+            if not isinstance(m, dict):
+                continue
+            media_id = m.get('id') or share_id
+            title = m.get('title') or share_id
+            duration = parse_duration(m.get('duration'))
+            thumbnails = m.get('thumbnails') or {}
+            thumbnail = thumbnails.get('poster') or thumbnails.get('image')
+
+            source = m.get('source') or {}
```

Comment:
1. Either assume that `title` will be present, or specialise IE._generic_title() to make s/t like `Video {m['id']} from vod2.infomaniak.com`. The ID alone adds nothing, since it's not meaningful (not like `conference-a-dyo`, say).
2. Metadata extraction can be done in the return expression.
3. In the previous suggestion the list was filtered for items with a `source` `dict`.
```suggestion
            source = m['source']
```

Reply:
Makes sense. I now treat source as required (the traversal filters for entries with a source dict) and simplified the code accordingly. Metadata extraction is also consolidated rather than being scattered across temporary variables.